lib/Sema/CSDiagnostics.cpp

```diff
@@ -4057,8 +4057,9 @@ bool MissingArgumentsFailure::diagnoseSingleMissingArgument() const {
         .fixItInsert(insertLoc, insertText.str());
   } else if (isPropertyWrapperInitialization()) {
     auto *TE = cast<TypeExpr>(fnExpr);
+    auto instanceTy = getConstraintSystem().getInstanceType(TE);
```

Comment:
The problem here is that we would like stop applying solution back to the constraint system before diagnosing so we can't really use `ConstraintSystem::getInstanceType` here. I think it would be okay to add a new `getInstanceType(TypeExpr *)` method to the `FailureDiagnostic` which would retrieve types based on solution associated with diagnostic.

Reply:
Ah, OK. I'm just going to put them back the way I found them.

Reply:
The problem is - apparently as-is is incorrect as well, could you please fix that since you are in the area? :)